apps/mobile/src/features/account/account.tsx

```diff
@@ -1,40 +1,47 @@
-import { Balance } from '@/components/balance/balance';
 import { NakedHeader } from '@/components/headers/naked-header';
 import { FetchError } from '@/components/loading/error';
 import { PageLayout } from '@/components/page/page.layout';
 import { AccountOverview } from '@/features/account/components/account-overview-card';
 import { ActivityWidget } from '@/features/activity/activity-widget';
 import { AccountBalances } from '@/features/balances/balances';
 import { BalancesWidget } from '@/features/balances/balances-widget';
-import { TokenBalance } from '@/features/balances/token-balance';
 import { Collectibles, CollectiblesWidget, hasCollectibles } from '@/features/collectibles';
 import { useCollectiblesFlag } from '@/features/feature-flags';
 import { NetworkBadge } from '@/features/settings/network-badge';
 import { useAccountActivity } from '@/queries/activity/account-activity.query';
+import { useAccountBalance } from '@/queries/balance/account-balance.query';
 import { useAccountCollectibles } from '@/queries/collectibles/account-collectibles.query';
 import { AppRoutes } from '@/routes';
+import { AccountLookup } from '@/shared/types';
 import { Account as AccountType } from '@/store/accounts/accounts';
 import { t } from '@lingui/macro';
 import { router } from 'expo-router';
 
 import { Box, SettingsGearIcon } from '@leather.io/ui/native';
 
-interface AccountProps {
+import { AccountBalance } from '../balances/total-balance';
+
+interface AccountProps extends AccountLookup {
   account: AccountType;
   walletName: string;
-  balance: TokenBalance;
-  isLoading: boolean;
-  isError: boolean;
 }
 
-export function Account({ account, walletName, balance, isLoading, isError }: AccountProps) {
-  const activity = useAccountActivity(account.fingerprint, account.accountIndex);
-  const collectibles = useAccountCollectibles(account.fingerprint, account.accountIndex);
+export function Account({ account, walletName }: AccountProps) {
+  const { fingerprint, accountIndex, id, name, icon } = account;
+  const { totalBalance } = useAccountBalance({
+    fingerprint,
+    accountIndex,
+  });
+  const activity = useAccountActivity(fingerprint, accountIndex);
+  const collectibles = useAccountCollectibles(fingerprint, accountIndex);
   const releaseCollectibles = useCollectiblesFlag();
+
+  const isLoadingTotalBalance = totalBalance.state === 'loading';
```

Comment:
We need to access `totalBalance.state` here to know whether to show a warning or not